Lib/test/test__opcode.py

```diff
@@ -64,5 +64,27 @@ def test_stack_effect_jump(self):
                     self.assertEqual(nojump, common)
 
 
+class SpecializationStatsTests(unittest.TestCase):
+    def test_specialization_stats(self):
+        STAT_NAMES = ['specialization_success', 'specialization_failure',
+                      'hit', 'deferred', 'miss', 'deopt', 'unquickened']
```

Comment:
Can we derive this list from elsewhere?
Maybe move this list to `opcode.py` and generate it much like we generate the specialized instructions and jump tables?

Reply:
I agree with this suggestion and have started working on it. It's a fairly large change to the existing code, so might be appropriate to do in a separate PR.

Reply:
There is a slight complication with this due to NEED_OPCODE_JUMP_TABLES. Is that temporary?

The issue is this: 
compile.c does
```
#define NEED_OPCODE_JUMP_TABLES
#include "opcode.h"               // EXTENDED_ARG
```


so if I want specialise.c to include opcode.h I need to put the define there as well. But that is beginning to be messy. So maybe we move the define into opcode.h so that it can be included more than once?

Reply:
Another option is that I add a specialize.h file with specialization related definitions, and a Tools/scripts/generate_specialize.h that creates it.